components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+                                 // output
+                                 MAMAci::view_2d aitken_dry_dia) {
+  const int aitken_idx = static_cast<int>(mam4::ModeIndex::Aitken);
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, top_lev, nlev), KOKKOS_LAMBDA(int kk) {
+        aitken_dry_dia(icol, kk) = dgnum(icol, aitken_idx, kk);
+      });
+}
+void compute_aitken_dry_diameter(haero::ThreadTeamPolicy team_policy,
+                                 const MAMAci::const_view_3d dgnum,
+                                 const int top_lev, const int nlev,
+                                 // output
+                                 MAMAci::view_2d aitken_dry_dia) {
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        compute_aitken_dry_diameter(team, dgnum, icol, top_lev, nlev,
+                                    // output
+                                    aitken_dry_dia);
+      });
+}
+
+void compute_nucleate_ice_tendencies(
+    const mam4::NucleateIce &nucleate_ice, haero::ThreadTeamPolicy team_policy,
+    const mam_coupling::DryAtmosphere &dry_atmosphere,
+    const mam_coupling::AerosolState &dry_aero, const MAMAci::view_2d wsubice,
+    const MAMAci::view_2d aitken_dry_dia, const int nlev, const double dt,
+    // output
+    MAMAci::view_2d nihf, MAMAci::view_2d niim, MAMAci::view_2d nidep,
+    MAMAci::view_2d nimey, MAMAci::view_2d naai_hom,
+    // ## output used by other processes ##
+    MAMAci::view_2d naai) {
+  //-------------------------------------------------------------
+  // Get number of activated aerosol for ice nucleation (naai)
+  // from ice nucleation
+  //-------------------------------------------------------------
+
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        //---------------------------------------------------------------------
+        //   Set up surface, pronostics atmosphere, diagnostics, and tendencies
+        //   classes.
+        //---------------------------------------------------------------------
+
+        // For compute_tendecies interface only, this structure is empty
+        haero::Surface surf{};
+
+        // Store interstitial and cld borne aerosols in "progrs" struture
+        mam4::Prognostics progs =
+            mam_coupling::aerosols_for_column(dry_aero, icol);
+
+        // Store atmopsheric vars (Tmid, Pmid, cloud fraction, qv, wsubmin)
+        haero::Atmosphere haero_atm =
+            atmosphere_for_column(dry_atmosphere, icol);
+
+        // Update the updraft velocity needed by nucleation to be "wsubice"
+        // in the haero_atm object
+        haero_atm.updraft_vel_ice_nucleation = ekat::subview(wsubice, icol);
+
+        // All the output from this process is diagnotics; creates "diags" with
+        // nlev column length
+        mam4::Diagnostics diags(nlev);
```

Comment:
I wonder if we should remove `mam4::Diagnostics `or replace by a struct with diagnostic variables for each process? We have many variables in` mam4::Diagnostics` and I find sometime difficult to know why variables I need to allocate.

Reply:
We will eventually do this cleanup. For now, let's keep the existing data structures as is.